src/NuGet.Core/NuGet.Build.Tasks/RestoreTaskEx.cs

```diff
@@ -0,0 +1,222 @@
+
+        public bool HideWarningsAndErrors { get; set; }
+
+        public bool IgnoreFailedSources { get; set; }
+
+        public bool Interactive { get; set; }
+
+        [Required]
+        public string MSBuildBinPath { get; set; }
+
+        public bool NoCache { get; set; }
+
+        [Required]
+        public string ProjectFullPath { get; set; }
+
+        public bool Recursive { get; set; }
+
+        public string SolutionPath { get; set; }
+
+        public void Cancel() => _cancellationTokenSource.Cancel();
+
+        public void Dispose()
+        {
+            _cancellationTokenSource.Dispose();
+        }
+
+        public override bool Execute()
+        {
+            var globalProperties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+            {
+                ["ExcludeRestorePackageImports"] = "true"
+            };
+
+            var properties = new Dictionary<string, string>(8, StringComparer.OrdinalIgnoreCase);
+
+            AddPropertyIfTrue(properties, nameof(DisableParallel), DisableParallel);
+            AddPropertyIfTrue(properties, nameof(Force), Force);
+            AddPropertyIfTrue(properties, nameof(ForceEvaluate), ForceEvaluate);
+            AddPropertyIfTrue(properties, nameof(HideWarningsAndErrors), HideWarningsAndErrors);
+            AddPropertyIfTrue(properties, nameof(IgnoreFailedSources), IgnoreFailedSources);
+            AddPropertyIfTrue(properties, nameof(Interactive), Interactive);
+            AddPropertyIfTrue(properties, nameof(NoCache), NoCache);
+            AddPropertyIfTrue(properties, nameof(Recursive), Recursive);
+
+            string entryProjectFile =
+                !string.IsNullOrWhiteSpace(SolutionPath) && !string.Equals(SolutionPath, "*Undefined*", StringComparison.OrdinalIgnoreCase)
+                    ? SolutionPath
+                    : ProjectFullPath;
+            try
+            {
+                var thisAssembly = new FileInfo(Assembly.GetExecutingAssembly().Location);
+
+                var startInfo = new ProcessStartInfo
+                {
+#if NETFRAMEWORK
+                    Arguments = $"\"{string.Join(";", properties.Select(i => $"{i.Key}={i.Value}"))}\" \"{Path.Combine(MSBuildBinPath, "MSBuild.exe")}\" \"{string.Join(";", entryProjectFile)}\" \"{string.Join(";", globalProperties.Select(i => $"{i.Key}={i.Value}"))}\"",
+#else
+                    Arguments = $"\"{Path.Combine(thisAssembly.DirectoryName, Path.ChangeExtension(thisAssembly.Name, ".Console.dll"))}\" \"{string.Join(";", properties.Select(i => $"{i.Key}={i.Value}"))}\" \"{Path.Combine(MSBuildBinPath, "MSBuild.dll")}\" \"{string.Join(";", entryProjectFile)}\" \"{string.Join(";", globalProperties.Select(i => $"{i.Key}={i.Value}"))}\"",
+#endif
+                    CreateNoWindow = true,
+#if NETFRAMEWORK
+                    FileName = Path.Combine(thisAssembly.DirectoryName, Path.ChangeExtension(thisAssembly.Name, ".Console.exe")),
```

Comment:
The two assemblies(NuGet.Build.Tasks and NuGet.Build.Tasks.Console( need to stay in syn regarding with the name. Do we need an unit test?